pkg/grafana/dashboards.go

```diff
@@ -1,296 +1,115 @@
 package grafana
 
 import (
-	"bytes"
-	"encoding/json"
 	"errors"
 	"fmt"
-	"io"
-	"net/http"
+
+	gclient "github.com/grafana/grafana-openapi-client-go/client"
+	"github.com/grafana/grafana-openapi-client-go/client/dashboards"
+	"github.com/grafana/grafana-openapi-client-go/client/search"
+	"github.com/grafana/grafana-openapi-client-go/client/snapshots"
+	"github.com/grafana/grafana-openapi-client-go/models"
 
 	"github.com/grafana/grizzly/pkg/grizzly"
 )
 
-// getRemoteDashboard retrieves a dashboard object from Grafana
-func getRemoteDashboard(uid string) (*grizzly.Resource, error) {
-	client := new(http.Client)
-	grafanaURL, err := getGrafanaURL("api/dashboards/uid/" + uid)
-	if err != nil {
-		return nil, err
-	}
+// Moved from utils.go
+const generalFolderId = 0
+const generalFolderUID = "general"
 
-	req, err := http.NewRequest("GET", grafanaURL, nil)
-	if err != nil {
-		return nil, err
-	}
+// Losing id, typeLogoUrl, version, withCredentials
 
-	if grafanaToken, ok := getGrafanaToken(); ok {
-		req.Header.Set("Authorization", "Bearer "+grafanaToken)
-	}
-	resp, err := client.Do(req)
+// getRemoteDashboard retrieves a dashboard object from Grafana
+func getRemoteDashboard(client *gclient.GrafanaHTTPAPI, uid string) (*grizzly.Resource, error) {
+	params := dashboards.NewGetDashboardByUIDParams().WithUID(uid)
+	dashboardOk, err := client.Dashboards.GetDashboardByUID(params, nil)
 	if err != nil {
+		var gErr *dashboards.GetDashboardByUIDNotFound
+		if errors.As(err, &gErr) {
+			return nil, grizzly.ErrNotFound
+		}
 		return nil, err
 	}
-	defer resp.Body.Close()
+	dashboard := dashboardOk.GetPayload()
 
-	switch {
-	case resp.StatusCode == http.StatusNotFound:
-		return nil, grizzly.ErrNotFound
-	case resp.StatusCode >= 400:
-		return nil, errors.New(resp.Status)
-	}
-
-	data, err := io.ReadAll(resp.Body)
+	// TODO: Turn spec into a real models.DashboardFullWithMeta object
+	spec, err := structToMap(dashboard.Dashboard)
```

Comment:
We have github.com/mitchellh/mapstructure that can do this too.

Reply:
I'm not sure if it's as easy as that, see 👉🏻  https://github.com/grafana/grizzly/pull/257#discussion_r1352778493

Reply:
thanks for these pointers, can go with the code in the PR then.